### ISSUE TITLE
Improving startup time

### ISSUE THREAD

**Issue by jayjlawrence** [joshukraine/dotfiles#19]
Thanks for the great work here. I found that after installing my zsh startup was not very attractive. Using a zsh profiler I found that the biggest time sucker is "export RUBY_CONFIGURE_OPTS" in .zshrc line 22.  ruby-build uses this when installing Ruby under rbenv and asdf (at least).

As along as we're not straying from the Home-brew supplied libraries for headline libyaml and gdbm then this value can be safely put inline as 

```
$ RUBY_CONFIGURE_OPTS="--with-opt-dir=`brew --prefix openssl`:`brew --prefix readline`:`brew --prefix libyaml`:`brew --prefix gdbm`"
```

It gives a noticeable improvement to startup time. Then there's some script for Spotify in the osx plugin for zsh that sucks. 

https://esham.io/2018/02/zsh-profiling was really helpful to profile the startup time ... 

If you want it could create a flaggable startup time profile feature for this project.

**Comment by jayjlawrence** [comment]
See #20

**Comment by joshukraine** [comment]
Wow, thank you for this! I've noticed the startup lag lately and been annoyed by it. Just haven't had the time to research. I tried out your zsh profiler solution along with the update to RUBY_CONFIGURE_OPTS and that made a huge difference!

I've merged #22 and #24. 😃👍🏻
